### PR TITLE
Fix link from generated configuration file to documentation

### PR DESCRIPTION
**Pull request by fiveop**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Review by reivilibre — APPROVED** [review]

**Comment by reivilibre** [comment]
Nicely spotted! Thanks for sending this in!

**Comment by fiveop** [comment]
I just found [sample_config.yaml](https://github.com/matrix-org/synapse/blob/master/docs/sample_config.yaml). How do you keep the code in `synapse/config/*.py` files in sync with this sample config? Or if you don't, I should make the same change in the sample file.

**Comment by clokep** [comment]
> I just found [sample_config.yaml](https://github.com/matrix-org/synapse/blob/master/docs/sample_config.yaml?rgh-link-date=2022-01-04T11%3A17%3A06Z). How do you keep the code in `synapse/config/*.py` files in sync with this sample config? Or if you don't, I should make the same change in the sample file.

You can run `./scripts-dev/generate_sample_config` to regenerate the sample config, which should port the changes over.

File: changelog.d/11678.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix the documentation link in newly-generated configuration files.
```
Comment (clokep) [inline changelog.d/11678.bugfix:1]: Should this be a `.doc` file?

Comment (fiveop) [reply changelog.d/11678.bugfix:1]: I was not sure. It is in code that generates something (among other things a broken link) that is essentially documentation. `.doc` was the other choice during my deliberation.

Comment (clokep) [reply changelog.d/11678.bugfix:1]: From the point of view of someone reading our release notes I think this just fixes documentation, so let's go with `.doc`! 👍

**Comment by reivilibre** [comment]
Thanks!

**Pull request merged**